packages/mui-base/src/utils/createSlot.tsx

```diff
@@ -0,0 +1,21 @@
+import * as React from 'react';
+
+export default function createSlot<
+  ComponentType extends React.JSXElementConstructor<React.ComponentProps<ComponentType>>,
+>(Component: ComponentType) {
+  return React.forwardRef<HTMLElement, React.ComponentProps<ComponentType>>(function Slot(
+    props,
+    ref,
+  ) {
+    // @ts-ignore ownerState is added by the owner component
+    const { ownerState, ...other } = props;
+
+    const rootProps = {
+      ref,
+      ...other,
+    } as unknown as JSX.IntrinsicAttributes &
```

Comment:
I couldn't find a better way to do this. I don't even understand why the types were complaining.

Reply:
I was able to rewrite it in a slightly simpler way:

```ts
export default function createSlot<ComponentType extends React.ElementType>(
  Component: ComponentType,
) {
  type Props = React.ComponentProps<ComponentType>;

  return React.forwardRef<HTMLElement, Props>(function Slot(props, ref) {
    const { ownerState, ...other } = props;
    return React.createElement<Props>(Component, {
      ...(other as Props),
      ref,
    });
  });
}
```

Reply:
Yep, much simpler! I was fighting with the types :D